src/opensles/AudioStreamBuffered.cpp

```diff
@@ -110,12 +110,22 @@ int64_t AudioStreamBuffered::predictNextCallbackTime() {
 
 // Common code for read/write.
 // @return Result::OK with frames read/written, or Result::Error*
-ResultWithValue<int32_t> AudioStreamBuffered::transfer(void *buffer,
-                                      int32_t numFrames,
-                                      int64_t timeoutNanoseconds) {
+ResultWithValue<int32_t> AudioStreamBuffered::transfer(
```

Comment:
I haven't looked at the Oboe codebase in a while, so I'm not up to date with styles. However, I wonder if this would be easier to read with as two functions, instead of one where we expect to pass a nullptr every time.

Reply:
Most of the transfer function is the same for read and write. I do not like like to have common code. I suppose I could pass a lambda but that would be getting tricky.

This seemed like an OK compromise where I could keep the benefit of common code and also use const to enforce data protection.

> one where we expect to pass a nullptr every time

Oboe users will never call this function. It is protected and strictly for internal use.